subsys/dm/dm.c

```diff
@@ -366,12 +369,34 @@ static void dm_reschedule(void)
 	}
 }
 
+static void calculation(void)
+{
+	if (IS_ENABLED(CONFIG_DM_MODULE_RPC_HOST)) {
+		struct dm_rpc_process_data *data;
+
+		data = dm_rpc_get_buffer(sizeof(*data));
+		nrf_dm_populate_report(&data->report);
+		bt_addr_le_copy(&data->bt_addr,
+				&timeslot_ctx.curr_req.dm_req.bt_addr);
+		dm_rpc_calc_and_process(data, sizeof(*data));
+	} else {
+		static nrf_dm_report_t report;
+
+		nrf_dm_populate_report(&report);
+		nrf_dm_calc(&report);
+		process_data(&report);
+
+		if (dm_context.cb->data_ready != NULL) {
+			dm_context.cb->data_ready(&result);
```

Comment:
Previously, this callback was called even if `dm_context.ranging_status` was `false`. Now, this function is called inside if in line 416. Is it ok?

Reply:
The current implementation seems to me more correct. 
The calculation and callback is called when `ranging_status` is `true`.